src/main/java/hudson/remoting/Launcher.java

```diff
@@ -568,6 +582,23 @@ private SSLSocketFactory getSSLSocketFactory()
         }
     }
 
+    private String readSecretFromFile(String secretFile) throws IOException {
+        BufferedReader reader = null;
+        String line = null;
+        try {
```

Comment:
What about using a try-with-resources? Is it not possible for any reason?

Reply:
Yes, that would be better.